jsonschema-hyperschema.xml

```diff
@@ -1539,6 +1389,221 @@ GET /foo/
+                    <t>Check that all required variables have a value</t>
+                    <t>Encode values into strings and fill out the template</t>
+                </list>
+            </t>
+            <figure>
+                <preamble>
+                    This is the high-level algorithm as pseudocode.  "T" comes from either
+                    "href" or "anchor" within the LDO, or from "base" in a containing schema.
+                    Pseudocode for each step follows.  "anchorOrHref" indicates which of the
+                    two started the process (since "base" is always resolved in order to finish
+                    resolving one or the other of those keywords).
+                </preamble>
+                <artwork>
+<![CDATA[
+templateData = populateDataFromInstance(T, ldo, instance)
+
+if anchorOrHref == "href" and ldo.hrefSchema exists:
+    inputData = acceptInput(ldo, instance, templateData)
+    for varname in inputData:
+        templateData[varname] = inputData[varname]
+
+for varname in ldo.templateRequired:
+    if not exists templateData[varname]
+        fatal("Missing required variable(s)")
+
+templateData = stringEncode(templateData)
+R = rfc6570ResolutionAlgorithm(T, templateData)
+]]>
+                </artwork>
+            </figure>
+
+            <section title="Populating template data from the instance">
+                <t>
+                    This step looks at various locations in the instance for variable values.
+                    For each variable:
+                    <list>
+                        <t>
+                            Use "templatePointers" to find a value if the variable
+                            appears in that keyword's value
+                        </t>
+                        <t>
+                            Otherwise, look for a property names matching the variable in
+                            the instance location to which the link is attached
+                        </t>
+                        <t>
+                            In either case, if there is a value at the location, put it in
+                            the template resolution data set
+                        </t>
+                    </list>
+                </t>
+                <figure>
+                    <artwork>
+<![CDATA[
+for varname in T:
+    varname = rfc3986PercentDecode(varname)
+    if varname in ldo.templatePointers:
+        valuePointer = templatePointers[varname]
+        if valuePointer is relative:
+            valuePointer = resolveRelative(attachmentPointer,
+                                           valuePointer)
+    else
+        valuePointer = attachmentPointer + "/" + varname
+
+    value = instance.valueAt(valuePointer)
+    if value is defined:
+        templateData[varname] = value
+]]>
+                    </artwork>
+                </figure>
+            </section>
+
+            <section title="Accepting input for template data">
```

Comment:
Might be worth mentioning that this step corresponds to the `acceptInput` call in this code block above.

Reply:
Really?  I have to say that "Accepting input" == "acceptInput"?  Even when it's in the right order and it's the same words already?  Let me think on if there is a better way to present this if it's that confusing as is.

Reply:
It's certainly not *confusing* now that you mention it. Perhaps I just read it without enough concentration... Let's leave this comment out for the moment.